packages/astro/src/runtime/client/dev-toolbar/apps/astro.ts

```diff
@@ -43,6 +44,17 @@ export default {
 				if (!integrationData) fetchIntegrationData();
 			}
 		});
+		eventTarget.addEventListener('placement-updated', (evt) => {
+			if (!(evt instanceof CustomEvent)) {
+				return;
+			}
+			const windowElement = canvas.querySelector('astro-dev-toolbar-window');
+			if (!windowElement) {
+				return;
+			}
+			const event: CustomEvent<{ placement: Placement }> = evt;
+			windowElement.placement = event.detail.placement;
+		});
```

Comment:
Since the logic is always the same, you can put this into a function, similar to `closeOnOutsideClick`

Reply:
That's a great idea! I've extract a `synchronizePlacementOnUpdate` function here: https://github.com/withastro/astro/pull/10591/commits/be69fc40bb6601fa0d26bb8dcf31f80330d3a520